core/src/test/java/org/infinispan/globalstate/ThreeNodeGlobalStatePartialRestartTest.java

```diff
@@ -0,0 +1,269 @@
+package org.infinispan.globalstate;
+
+import static org.infinispan.commons.test.CommonsTestingUtil.tmpDirectory;
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Map;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.test.Exceptions;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.StoreConfigurationBuilder;
+import org.infinispan.distribution.ch.ConsistentHash;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.partitionhandling.PartitionHandling;
+import org.infinispan.remoting.transport.jgroups.JGroupsAddress;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.topology.MissingMembersException;
+import org.infinispan.topology.PersistentUUID;
+import org.infinispan.topology.PersistentUUIDManager;
+import org.testng.annotations.Test;
+
+@Test(testName = "globalstate.ThreeNodeGlobalStatePartialRestartTest", groups = "functional")
+public class ThreeNodeGlobalStatePartialRestartTest extends AbstractGlobalStateRestartTest {
+
+   private PartitionHandling handling;
+   private CacheMode cacheMode;
+   private boolean sifs;
+   private boolean purge;
+
+   @Override
+   protected int getClusterSize() {
+      return 3;
+   }
+
+   @Override
+   protected void applyCacheManagerClusteringConfiguration(ConfigurationBuilder config) {
+      config.clustering().cacheMode(cacheMode).hash().numOwners(getClusterSize() - 1);
+      config.clustering().partitionHandling().whenSplit(handling);
+   }
+
+   @Override
+   protected void applyCacheManagerClusteringConfiguration(String id, ConfigurationBuilder config) {
+      applyCacheManagerClusteringConfiguration(config);
+
+      StoreConfigurationBuilder scb;
+      if (sifs) {
```

Comment:
The test shouldn't care if it is sifs or single file store. Please just use SIFS.